Dockerfile

```
@@ -1,61 +1,84 @@
 # Copyright DB InfraGO AG and contributors
 # SPDX-License-Identifier: Apache-2.0
 
-# Build frontend
-FROM node:20 AS build-frontend
-WORKDIR /app
-COPY frontend/package*.json ./
-RUN npm install
-COPY frontend/ ./
-RUN npm run build
-
-# Build backend
 FROM python:3.12-slim-bookworm
-WORKDIR /app
-
 USER root
-
-RUN apt-get update && \
-    apt-get install --yes --no-install-recommends \
-    git \
-    git-lfs \
-    libgirepository1.0-dev \
-    libcairo2-dev \
-    gir1.2-pango-1.0 \
-    graphviz \
-    nodejs \
-    npm && \
-    rm -rf /var/lib/apt/lists/*
-
-COPY ./capella_model_explorer ./capella_model_explorer
-COPY ./pyproject.toml ./
-COPY ./.git ./.git
-
-RUN pip install --no-cache-dir .
-COPY --from=build-frontend /app/dist/ ./frontend/dist/
-
-# Expose the port the app runs in
+WORKDIR /app
+ENV HOME=/home
+ENV PATH=$HOME/.local/bin:$PATH
+ENV MODEL_ENTRYPOINT=/model
+# Expose the port the app runs on
 EXPOSE 8000
 
-COPY ./templates /views
-
-ENV HOME=/home
+# install system pkgs {{{
+RUN apt-get update && \
+  apt-get install --yes --no-install-recommends \
+  curl \
+  git \
+  git-lfs \
+  gir1.2-pango-1.0 \
+  graphviz \
+  libcairo2-dev \
+  libgirepository1.0-dev \
+  npm && \
+  apt-get autoremove --yes && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/*
+# }}}
 
-COPY entrypoint.sh /entrypoint.sh
+# copy files {{{
+COPY capella_model_explorer capella_model_explorer
+COPY static static
+COPY templates templates
+COPY pyproject.toml ./
+COPY .git .git
+COPY package*.json ./
+# copy entrypoint.sh to root:
+COPY entrypoint.sh /
 RUN chmod +x /entrypoint.sh
-ENV MODEL_ENTRYPOINT=/model
-RUN chmod -R 777 ./frontend/dist/
+# }}}
 
-# Run script to get software version
-ENV MODE=production
-COPY frontend/fetch-version.py ./frontend/
-RUN python frontend/fetch-version.py && \
-    python -c "from capellambse_context_diagrams import install_elk; install_elk()"
+# misc setup {{{
+RUN git config --global --add safe.directory /model && \
+  git config --global --add safe.directory /model/.git && \
+  chmod -R 777 /app && \
+  chmod -R 777 /home
+# }}}
 
 # Run as non-root user per default
-RUN git config --global --add safe.directory /model && \
-    git config --global --add safe.directory /model/.git && \
-    chmod -R 777 /home
+USER 1000
+
+# install uv {{{
+RUN curl -Lo /tmp/install.sh https://astral.sh/uv/install.sh && \
+  chmod +x /tmp/install.sh && \
+  UV_NO_MODIFY_PATH=1 sh /tmp/install.sh && \
+  rm /tmp/install.sh
+# }}}
+
+# install app incl. its cli and install elk.js {{{
+RUN uv venv && \
+  # install app
+  uv pip install --no-cache-dir . && \
+  # Install elk.js automatically into a persistent local cache directory
+  # in order to prepare the elk.js execution environment ahead of time.
+  uv run python3 -c "from capellambse_context_diagrams import install_elk; install_elk()"
+# }}}
+
+# install Nodep kgs and build the CSS {{{
+RUN npm clean-install && uv run python3 -m capella_model_explorer build css
```

We don't even need a multi-stage build, we can just remove the `node_modules` right here. They're not needed after this step anymore, as far as I can tell.

```suggestion
RUN npm clean-install && uv run python3 -m capella_model_explorer build css && rm -rf node_modules
```